src/app/mail/mail-settings/mail-settings.component.ts

```diff
@@ -121,6 +126,41 @@ export class MailSettingsComponent implements OnInit, OnDestroy {
       'contact_email': ['', [Validators.pattern(VALID_EMAIL_REGEX)]],
       'password': ['', [Validators.required]]
     });
+    this.route.params.subscribe(
+      params => {
+        this.selectedTabQueryParams = params['id'];
+        this.navigateToTab(this.selectedTabQueryParams);
+      }
+    );
+  }
+
+  ngAfterViewInit() {
+   if (this.selectedTabQueryParams === 'dashboard-and-plans') {
```

Comment:
You don't need to make so many if else conditions, just use the same name in query params and tab set.